src/format/tests.rs

```diff
@@ -2623,6 +2623,22 @@ fn empty_lines() {
 "
     );
 
+    // Lines with only spaces are treated as empty
+    assert_format_rewrite!(
+        "pub fn main() {
+  let x = 1
+    
```

Comment:
Can we rewrite this as a line with `\n` at the end? An empty line with just spaces is hard to read and may be erased by some editors.

Reply:
Good point. I've rewritten and rebased. I wasn't sure the best way. I guess it could all be collapsed to one line but I've just collapsed the lines around it.